client/src/components/ToolsTab.tsx

```diff
@@ -175,6 +177,17 @@ const ToolsTab = ({
             <h3 className="font-semibold">
               {selectedTool ? selectedTool.name : "Select a tool"}
             </h3>
+            <div className="flex items-center gap-2">
+              {selectedTool && (
+                <IconDisplay
+                  icons={(selectedTool as WithIcons).icons}
+                  size="md"
+                />
+              )}
+              <h3 className="font-semibold">
+                {selectedTool ? selectedTool.name : "Select a tool"}
+              </h3>
+            </div>
```

Comment:
This section is leading to a doubling up of the tool name or "select a tool" message.

<img width="302" height="208" alt="Image" src="https://github.com/user-attachments/assets/6c4301db-725a-45ac-ae3c-462ed88dcde9" />

<img width="451" height="228" alt="Image" src="https://github.com/user-attachments/assets/dc8f2458-6533-4e5a-a173-4c49c5e8915e" />